File: pilha/estatica/Java/Pilha Genérica/Empilhavel.java
public interface Empilhavel<T> {
	//métodos principais
	void empilhar(T dado);	//C
	T espiar();				//R
	void atualizar(T dado);	//U
	T desempilhar();		//D

	//métodos auxiliares
	boolean estaCheia();
	boolean estaVazia();
	String imprimir();
}

File: pilha/Pilha Estática/Kotlin/Pilha/Empilhavel.kt
//https://www.javainuse.com/java2kot
//https://play.kotlinlang.org/
interface Empilhavel {
    // Metodos principais
    fun empilhar(dado: Any?)
    fun desempilhar(): Any?
    fun atualizar(dado: Any?)
    fun espiar(): Any?
    // Metodos auxiliares
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}